pkg/monitoring/prometheusagent/sharding/sharding_test.go

```diff
@@ -36,61 +50,88 @@ var tests = []struct {
 				timeSeries:        float64(2_000_001),
 				expected:          3,
 			},
+			{
+				currentShardCount: 0,
+				timeSeries:        float64(3_000_001),
+				expected:          4,
+			},
 		},
 	},
 	{
 		name:     "scale down",
 		strategy: defaultShardingStrategy,
 		cases: []testCase{
+			// Test cases when scale down threshold is hit
 			{
-				currentShardCount: 1,
-				timeSeries:        float64(1_000_001),
-				expected:          2,
+				currentShardCount: 2,
+				timeSeries:        float64(800_000),
+				expected:          1,
 			},
 			{
-				currentShardCount: 2,
-				timeSeries:        float64(999_999),
+				currentShardCount: 3,
+				timeSeries:        float64(1_800_000),
 				expected:          2,
 			},
+			{
+				currentShardCount: 4,
+				timeSeries:        float64(2_800_000),
+				expected:          3,
+			},
+			// Test cases when above scale down threshold
 			{
 				currentShardCount: 2,
 				timeSeries:        float64(800_001),
 				expected:          2,
 			},
 			{
-				currentShardCount: 2,
-				// 20% default threshold hit
-				timeSeries: float64(800_000),
-				expected:   1,
+				currentShardCount: 3,
+				timeSeries:        float64(1_800_001),
+				expected:          3,
+			},
+			{
+				currentShardCount: 4,
+				timeSeries:        float64(2_800_001),
+				expected:          4,
 			},
 		},
 	},
 	{
-		name:     "always defaults to 1",
+		name:     "keep current shards when no time series",
 		strategy: defaultShardingStrategy,
 		cases: []testCase{
 			{
 				currentShardCount: 0,
-				timeSeries:        float64(0),
+				timeSeries:        float64(-5),
 				expected:          1,
 			},
 			{
 				currentShardCount: 0,
-				timeSeries:        float64(-5),
+				timeSeries:        float64(0),
 				expected:          1,
 			},
+			{
+				currentShardCount: 1,
+				timeSeries:        float64(0),
+				expected:          1,
+			},
+			{
+				currentShardCount: 2,
+				timeSeries:        float64(0),
+				expected:          2,
+			},
+			{
+				currentShardCount: 3,
+				timeSeries:        float64(0),
+				expected:          3,
+			},
 		},
 	},
 }
 
 func TestShardComputationLogic(t *testing.T) {
```

Comment:
Why remove parallèl?

Reply:
There is no need for parallel execution here

Reply:
with t.Parallel

```
$ time go test ./pkg/monitoring/prometheusagent/sharding/
ok      github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent/sharding    (cached)
go test ./pkg/monitoring/prometheusagent/sharding/  0,11s user 0,05s system 192% cpu 0,084 total
```

without t.Parallel

```
$ time go test ./pkg/monitoring/prometheusagent/sharding/
ok      github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent/sharding    (cached)
go test ./pkg/monitoring/prometheusagent/sharding/  0,11s user 0,08s system 175% cpu 0,104 total
```

I see no speed improvement, and even then tests do not need such "optimization". t.Parallel is also meant to be used in order to spot race conditions rather than improving speed.

Reply:
Well I disagree that tests should not run on parallel for ci costs and Time but hère they are probably not relevant